Docker/scripts/gen.compose.py

```diff
@@ -71,23 +74,144 @@
       org.label-schema.group: "monitoring"
 
   # Does not work on MacOS, only Linux.
-  cadvisor:
-    image: gcr.io/cadvisor/cadvisor:v0.44.0
-    container_name: cadvisor
-    privileged: true
-    devices:
-      - /dev/kmsg:/dev/kmsg
+  # cadvisor:
+  #   image: gcr.io/cadvisor/cadvisor:v0.44.0
+  #   container_name: cadvisor
+  #   privileged: true
+  #   devices:
+  #     - /dev/kmsg:/dev/kmsg
+  #   volumes:
+  #     - /:/rootfs:ro
+  #     - /var/run:/var/run:ro
+  #     - /sys:/sys:ro
+  #     - /var/lib/docker:/var/lib/docker:ro
+  #     - /cgroup:/cgroup:ro
+  #   restart: unless-stopped
+  #   expose:
+  #     - 8080
+  #   labels:
+  #     org.label-schema.group: "monitoring"
+
+  read:
+    image: grafana/loki:2.5.0
+    command: "-config.file=/etc/loki/config.yaml -target=read"
+    ports:
+      - 3101:3100
+      - 7946
+      - 9095
     volumes:
-      - /:/rootfs:ro
-      - /var/run:/var/run:ro
-      - /sys:/sys:ro
-      - /var/lib/docker:/var/lib/docker:ro
-      #- /cgroup:/cgroup:ro #doesn't work on MacOS only for Linux
-    restart: unless-stopped
-    expose:
-      - 8080
-    labels:
-      org.label-schema.group: "monitoring"
+      - ./loki-config.yaml:/etc/loki/config.yaml
+    depends_on:
+      - minio
+    networks: &loki-dns
+      default:
+        aliases:
+          - loki
+
+  write:
+    image: grafana/loki:2.5.0
+    command: "-config.file=/etc/loki/config.yaml -target=write"
+    ports:
+      - 3102:3100
+      - 7946
+      - 9095
+    volumes:
+      - ./loki-config.yaml:/etc/loki/config.yaml
+    depends_on:
+      - minio
+    networks:
+      <<: *loki-dns
+
+  promtail:
+    image: grafana/promtail:2.5.0
+    volumes:
+      - ./promtail-local-config.yaml:/etc/promtail/config.yaml:ro
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /var/lib/docker/containers:/var/lib/docker/containers
+      - ./:/validators:ro
+    command: -config.file=/etc/promtail/config.yaml
+    depends_on:
+      - gateway
+
+  minio:
+    image: minio/minio
+    entrypoint:
+      - sh
+      - -euc
+      - |
+        mkdir -p /data/loki-data /data/loki-ruler && \
+        minio server /data
+    environment:
+      - MINIO_ACCESS_KEY=loki
+      - MINIO_SECRET_KEY=supersecret
+      - MINIO_PROMETHEUS_AUTH_TYPE=public
+      - MINIO_UPDATE=off
+    ports:
+      - 9000
+    volumes:
+      - ./data/minio:/data
+
+  gateway:
+    image: nginx:latest
+    depends_on:
+      - loki
```

Comment:
Ok seems that if I run like `docker-compose -f....` I run to the above issue (apparently this runs the docker compose V1). However, if I run like `docker compose....` (as you suggested on the README) then this will run the V2 which apparently has some differences. I might change the other parts of the README so we are aligned on this.